hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/core/GraphManager.java

```diff
@@ -2165,4 +2175,139 @@ public void accept(T t) {
         }
     }
 
+    private <T> void graphAddHandler(T response) {
+        List<String> names = this.metaManager
+                .extractGraphsFromResponse(response);
+        for (String graphName : names) {
+            String[] parts = graphName.split(DELIMITER);
+            if (parts.length < 2) {
+                LOG.error("The graph name format is incorrect: {}", graphName);
+                continue;
+            }
+            // If the current server is not registered to the DEFAULT schema,
+            // it will only receive graph creation events under the registered schemas.
+            if (!"DEFAULT".equals(this.serviceGraphSpace) &&
+                !parts[0].equals(this.serviceGraphSpace)) {
+                LOG.warn(String.format("Listen event: graph [%s] add was discarded because " +
+                                       "it did not belong to the graph space [%s] " +
+                                       "registered by the current server",
+                                       graphName, this.serviceGraphSpace));
+                // TODO: further confirmation is required
+                continue;
+            }
+            LOG.info("Accept graph add signal from etcd for {}", graphName);
+            if (this.graphs.containsKey(graphName) ||
+                this.creatingGraphs.contains(graphName)) {
+                this.creatingGraphs.remove(graphName);
+                continue;
+            }
+
+            LOG.info("Not exist in cache, Starting construct graph {}",
+                     graphName);
+            Map<String, Object> config =
+                    this.metaManager.getGraphConfig(parts[0], parts[1]);
+            if (config == null) {
+                LOG.error("The graph config not exist: {}", graphName);
+                continue;
+            }
+            Object objc = config.get("creator");
+            String creator = null == objc ?
+                             GraphSpace.DEFAULT_CREATOR_NAME :
+                             String.valueOf(objc);
+
+            // Create graph without init
+            try {
+                HugeGraph graph;
+                // TODO: add alias graph
+                graph = this.createGraph(parts[0], parts[1], creator, config, false);
```

Comment:
‼️ **Critical: 异常处理不完整**

在 `graphAddHandler` 中，如果图创建失败且 `startIgnoreSingleGraphError=false`，会抛出异常。但在此之前，`creatingGraphs` 集合没有被清理，可能导致内存泄漏和状态不一致。

**建议**：使用 try-finally 确保清理逻辑：
```suggestion
try {
    HugeGraph graph;
    graph = this.createGraph(parts[0], parts[1], creator, config, false);
    LOG.info("Add graph space:{} graph:{}", parts[0], parts[1]);
    boolean grpcThread = Thread.currentThread().getName().contains("grpc");
    if (grpcThread) {
        HugeGraphAuthProxy.setAdmin();
    }
    graph.started(true);
    if (graph.tx().isOpen()) {
        graph.tx().close();
    }
} catch (HugeException e) {
    this.creatingGraphs.remove(graphName); // 确保清理
    if (!this.startIgnoreSingleGraphError) {
        throw e;
    }
    LOG.error(String.format(
            "Failed to create graph '%s'", graphName), e);
}
```